### ISSUE TITLE
Writing custom audit trail entries

### ISSUE THREAD

**Issue by Mumfrey** [cpw/modlauncher#38]
I've been looking at having Mixin write more information to the audit trail to document its activities and have hit a small snag. While the audit trail is easy to retrieve, it's only possible to write the 3 existing record types via `addReason`, `addPluginAuditTrail` and `addTransformerAuditTrail`.

Whilst transformer actions can be traced reasonably, it would be useful if launch plugins could be supplied with a delegate to add their own (scoped) audit trail entries which could be recorded under (for example) `PLUGIN_ACTIVITY` as `Type` or simply recorded under the existing `PLUGIN` type with plugin-specific detail (`D`?) included as the `context`.

For example if I apply 3 mixins to a class, being able to do something like 
```java
pluginAuditTrail.addActivity(className, "apply", "mixin.name");
// or maybe 
pluginAuditTrail.addActivity(className, "+", "mixin.name"); // shorter is better?
```
*(edited because I forgot class name argument)*

ideally we'd end up with something like this in the audit trail:
```
net.minecraft.client.gui.screen.MainMenuScreen {
   PLUGIN:mixin:A
   PLUGIN:mixin:D:+:mixins.foo.bar.ScreenMixin1
   PLUGIN:mixin:D:+:com.blah.ScreenMixin
}
```
I don't particularly want to go crazy with detail, but just get some of the more significant events recorded in the audit trail in a consistent manner.

**Comment by Mumfrey** [comment]
Herewith my preliminary scribbles for audit trail wiring:

[`bd5179d Some audit trail wiring for ModLauncher, for science `](https://github.com/SpongePowered/Mixin/commit/bd5179d309299ce2915358e4ba38a9fbdbbb9487)

I've modelled the delegate as `BiConsumer<String, String[]>` for now (as `className`, `args`) but you get the idea.

**Comment by cpw** [comment]
@Mumfrey didn't we deal with this issue?

**Comment by Mumfrey** [comment]
Yes, resolved in 48e1926a3bd86c16bc505961dee21a082196de84
